File: Modules/ServerTypes/Articles/CategoryRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface CategoryRow {
    CategoryId?: number;
    Title?: string;
}

export abstract class CategoryRow {
    static readonly idProperty = 'CategoryId';
    static readonly nameProperty = 'Title';
    static readonly localTextPrefix = 'Articles.Category';
    static readonly lookupKey = 'Articles.Category';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<CategoryRow>('Articles.Category') }
    static async getLookupAsync() { return getLookupAsync<CategoryRow>('Articles.Category') }

    static readonly deletePermission = 'Articles:Categories';
    static readonly insertPermission = 'Articles:Categories';
    static readonly readPermission = 'Articles:Categories';
    static readonly updatePermission = 'Articles:Categories';

    static readonly Fields = fieldsProxy<CategoryRow>();
}

File: Modules/Articles/Category/CategoryRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace Dashboard.Articles;

[ConnectionKey("Default"), Module("Articles"), TableName("[serene].[Category]")]
[DisplayName("Category"), InstanceName("Category")]
[ReadPermission(PermissionKeys.Categories)]
[ModifyPermission(PermissionKeys.Categories)]
[InsertPermission(PermissionKeys.Categories)]
[LookupScript("Articles.Category", Permission = PermissionKeys.Categories, Expiration = -1)]
public sealed class CategoryRow : Row<CategoryRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Category Id"), Identity, IdProperty]
    public int? CategoryId
    {
        get => fields.CategoryId[this];
        set => fields.CategoryId[this] = value;
    }

    [DisplayName("Title"), Size(200), NotNull, QuickSearch, NameProperty]
    public string Title
    {
        get => fields.Title[this];
        set => fields.Title[this] = value;
    }

    public class RowFields : RowFieldsBase
    {
        public Int32Field CategoryId;
        public StringField Title;

    }
}